### ISSUE TITLE
Training data coming up as null and circular errors

### ISSUE THREAD

**Issue by ruhman** [openai/gym#594]
Hi!
When using `env = gym.wrappers.Monitor(env, args.out,force=True)`

I get the error:
`You didn't have any recorded training data in /Users/daniruhman/Downloads/atariPyDQN/gym-out/SpaceInvaders-v0/. Once you've used 'env.monitor.start(training_dir)' to start recording, you need to actually run some rollouts. Please join the community chat on https://gym.openai.com if you have any issues. `

However the directory is not empty. The .stats.json file is for some reason coming up:
`{"timestamps": [], "initial_reset_timestamp": null, "episode_types": [], "episode_lengths": [], "episode_rewards": []} `

And, if I try to run `env.monitor.start(training_dir)` as suggested by the error, I get:

`Error: env.monitor has been deprecated as of 12/23/2016. Remove your call to `env.monitor.start(directory)` and instead wrap your env with env = gym.wrappers.Monitor(env, directory) to record data. `

Which tells me to use `env = gym.wrappers.Monitor(env, directory)`, my original command.

TL;DR

Circular errors, one leads to the other. Training data coming up null.

If you want, you can checkout the code at: https://github.com/ruhman/atariPyDQN

**Comment by tlbtlbtlb** [comment]
You're giving the un-wrapped environment to the DQN agent at https://github.com/ruhman/atariPyDQN/blob/master/atari.py#L71
You should wrap it first by moving this line above it: https://github.com/ruhman/atariPyDQN/blob/master/atari.py#L77

**Comment by ruhman** [comment]
Thanks! I have tried that.
Unfortunatelly that starts giving errors like 
`AttributeError: 'Monitor' object has no attribute 'act'` etc. since it redefines de env as a monitor instead of the custom environment. I think I can update it and fix it tho. Will have a look tomorrow and post.
Still, the main thing about this issue is that the error message `You didn't have any recorded training data in /Users/daniruhman/Downloads/atariPyDQN/gym-out/SpaceInvaders-v0/. Once you've used 'env.monitor.start(training_dir)' to start recording, you need to actually run some rollouts. Please join the community chat on https://gym.openai.com if you have any issues.` should be updated to reflect the new commands, no?
I will do a PR later.
Thanks!

**Comment by tlbtlbtlb** [comment]
You should only be using `.step` and `.reset` functions on the env.

If you need to peek under the covers, you can get to it with `env.unwrapped.act(...)`. But then the monitor won't work correctly, since it doesn't see the actions (it wraps the `.step` call to get them).

The error message is wrong as you say, thanks for the fix.
